### DISCUSSION TITLE
How does the sync with kubectl work actually

### DISCUSSION THREAD

**Discussion by jabbrwcky** [argoproj/argo-cd#17036] [Q&A]
Hi,

there is a lot of information about applying resources at https://argo-cd.readthedocs.io/en/stable/user-guide/sync-kubectl/#sync-applications-with-kubectl

IMO it lacks the interesting part, e.g. like how to do some common thing like 
"Install the CRDs before installing the operator":

```shell
# First, need to install the CRDs manually.
kubectl apply -f https://github.com/nats-io/nack/releases/latest/download/crds.yml
```

The documentation specifies an example for kubectl apply, but sadly omits *how* the file `<apply-file>` is specified in the Application. 

Could anyone explain how to do this (I would be open to open a PR for the documentation after understanding how to achieve this)

**Comment by jgwest** [comment]
For the specific use case you mention, you can use [sync waves](https://argo-cd.readthedocs.io/en/stable/user-guide/sync-waves/) to control the order in which resources are applied. You may [also need to disable dry run](https://github.com/argoproj/argo-cd/discussions/11883#discussion-4723312) if your GitOps repository also contains the CRs themselves (for example you are installing Postgresql operator, then using `PostgreSQL` CR, in the same Argo CD Application)
